### PR TITLE
[manual] add set op & fix many typos

### LINKED ISSUE #765: [Doc]Name Doc Files
It's better for us to unify the way to name doc files. To me, doc files should be named as `how-to-contribute`, worlds should be separated with `-` rather than `_`.

### PR DESCRIPTION
**Pull request by whitewum**
Add set ops
close #765

### CODE REVIEW

**Comment by whitewum** [comment]
group by, limit doc will be given after their pr merged

File: docs/manual-doc/functions-and-operators/set-operations.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+## `UNION`, `UNION DISTINCT`, and `UNION ALL`
+
+Operation `UNION DISTINCT` (or by short `UNION`) returns the union of two sets A and B (denoted by $A \bigcup B$ in mathematics), with the distinct element belongs to set A or set B, or both.
+
+Meanwhile, operation `UNION ALL` returns the union set with duplicated elements.
+
+```
+<left> UNION [DISTINCT | ALL] <right>
+```
+where `<left>` and `<right>` must have the same number of columns and pair-wise data types.
```
Comment (darionyaphet) [inline docs/manual-doc/functions-and-operators/set-operations.md]: First letter should be upper case ?

File: docs/manual-doc/statement-syntax/data-manipulation-statements/pipe-syntax.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+### Examples
+
+```
+GO FROM 201 OVER edge_serve | GO FROM $-.id OVER edge_fans | GO FROM $-.id ...
+
+GO FROM 100 OVER like YIELD like._dst AS dstid, $$.player.name AS Name  | GO FROM $-.dstid OVER like YIELD like._dst, like.likeness, $-.Name
+```
+
+The alias name right after placeholder `$-.` must be either exactly `id` or defined in the previews statement (e.g., `distid` or `Name` as show in the above example).
```
Comment (CPWstatic) [inline docs/manual-doc/statement-syntax/data-manipulation-statements/pipe-syntax.md]: The dest `id` would be given by default when `yield` is not declared. If `yield` is declared explicitly, `id` would not be given unless user declares explicitly.

Comment (whitewum) [reply docs/manual-doc/statement-syntax/data-manipulation-statements/pipe-syntax.md]: After some consideration, I choose not to mention this short-cut-syntax, as too much for the first time to read

File: docs/manual-doc/data-types/numeric-types.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Two basic data types are supported by `Nebula` -- int64 & double.
+
+The range for int64 is [-9223372036854775808, 9223372036854775807]. There are no upper and lower bound for double.
```
Comment (jude-zhu) [inline docs/manual-doc/data-types/numeric-types.md]: no upper or lower bound

File: docs/manual-doc/functions-and-operators/order-by-function.md [hunk 4]
```diff
@@ -6,7 +6,7 @@ And it can only be used in the `PIEP`-syntax ("|")
 ```
 | ORDER BY <prop> ASC | DESC [, <prop> ASC | DESC ...] 
 ```
-By default, `ORDER BY` sorts in ascending order if no ASC or DESC is given.
+By default, `ORDER BY` sorts in ascending order if no `ASC` or `DESC` is given.
```
Comment (jude-zhu) [inline docs/manual-doc/functions-and-operators/order-by-function.md]: `ORDER BY` sorts the records in ascending...

File: docs/manual-doc/data-types/numeric-types.md (hunk 3 above)
Comment (whitewum) [reply docs/manual-doc/data-types/numeric-types.md]: we 'll start another pr to fix those typos and minor errors.

**Comment by whitewum** [comment]
There is another pr working on the minor errors by Fan & Yinyin.
So, I'll let the minor errors un-corrected in this pr -- in case duplicate work.
FYI.

File: docs/manual-doc/functions-and-operators/set-operations.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+```
+GO FROM 1 OVER e1 YIELD e1._dst AS id, e1.prop1 AS left.1, $$.tag.prop2 AS left.2
+UNION /* DISTINCT */
+GO FROM 2,3 OVER e1 YIELD e1._dst AS id, e1.prop1 AS right.1, $$.tag.prop2 AS right.2
+```
+
+will return as follows:
+```
+==========================
```
Comment (darionyaphet) [inline docs/manual-doc/functions-and-operators/set-operations.md]: remove the extra `=` and `-`

File: docs/manual-doc/functions-and-operators/set-operations.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+
+### Example
+
+The following statement
+```
+GO FROM 1 OVER e1 \
+UNION \
+GO FROM 2 OVER e1
+```
+return the distinct neighbors vertex id of `1` and `2` (along with edge `e1`).
```
Comment (jude-zhu) [inline docs/manual-doc/functions-and-operators/set-operations.md]: return the distinct neighbors' vertex id of `1` and `2`?

File: docs/manual-doc/functions-and-operators/set-operations.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+
+While
+```
+GO FROM 1 OVER e1 \
+UNION ALL\
+GO FROM 2 OVER e1
+```
+returns all the neighbors vertex id of `1` and `2`, with all duplications.
```
Comment (jude-zhu) [inline docs/manual-doc/functions-and-operators/set-operations.md]: ditto

File: docs/manual-doc/functions-and-operators/set-operations.md [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+| 104 |    2   |    2   |    -- line 2
+--------------------------
+| 215 |    4   |    3   |    -- line 3
+--------------------------
+```
+
+## INTERSECT
+
+Operation `INTERSECT` returns the intersection of two sets A and B (denoted by $A \bigcap B$), if the elements belongs both to set A and set B.
+
```
Comment (jude-zhu) [inline docs/manual-doc/functions-and-operators/set-operations.md]: Operation => operator? 
I think "_if the elements belongs both to set A and set B_" can be deleted. btw. belong

File: docs/manual-doc/functions-and-operators/set-operations.md [hunk 9]
```diff
@@ -0,0 +1,166 @@
+# Set Operations (`UNION`, `INTERSECT`, and `MINUS`)
+
+## `UNION`, `UNION DISTINCT`, and `UNION ALL`
+
+Operation `UNION DISTINCT` (or by short `UNION`) returns the union of two sets A and B (denoted by $A \bigcup B$ in mathematics), with the distinct element belongs to set A or set B, or both.
```
Comment (jude-zhu) [inline docs/manual-doc/functions-and-operators/set-operations.md]: Operation => operator?

File: docs/manual-doc/functions-and-operators/set-operations.md [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+==========================
+| id  | left.1 | left.2 |
+==========================
+| 104 |    1   |    2   |    -- line 1
+--------------------------
+```
+
+## MINUS
+
+The set subtraction (or difference), $A - B$, consists of elements that are in A but not in B. So the operation order matters.
```
Comment (jude-zhu) [inline docs/manual-doc/functions-and-operators/set-operations.md]: '$A - B$' is not supported.

File: docs/manual-doc/functions-and-operators/set-operations.md [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+
+```
+==========================
+| id  | left.1 | left.2  |
+==========================
+| 215 |    4   |    3    |     -- line 3
+--------------------------
+```
+
+And if we reserve the `MINUS` order
```
Comment (jude-zhu) [inline docs/manual-doc/functions-and-operators/set-operations.md]: reserve  => reverse

File: docs/manual-doc/language-structure/property-reference.md [hunk 12]
```diff
@@ -42,13 +42,13 @@ edge_type.edge_prop
...
 
-for example,
+For example,
 
 ```
 GO FROM 1 OVER e1 YIELD e1.prop1
 ```
 
-### for build-in property
+### For build-in property
```
Comment (jude-zhu) [inline docs/manual-doc/language-structure/property-reference.md]: properties

File: docs/manual-doc/functions-and-operators/set-operations.md [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+
+```
+GO FROM 2,3 OVER e1 YIELD e1._dst AS id, e1.prop1 AS right.1, $$.tag.prop2 AS right.2
+MINUS
+GO FROM 1 OVER e1 YIELD e1._dst AS id, e1.prop1 AS left.1, $$.tag.prop2 AS left.2
+```
+
+will come out
```
Comment (jude-zhu) [inline docs/manual-doc/functions-and-operators/set-operations.md]: will come out => results in?

File: docs/manual-doc/language-structure/property-reference.md [hunk 14]
```diff
@@ -59,10 +59,10 @@ There are four build-in properties in the edge:
...
 
-for example,
+For example,
 
 ```
 GO FROM 1 OVER e1 YIELD e1._src as startVID /* which is, 1 */, e1._dst as endVID
 ```
 
-This statement returns all the neighborhoods of `1` over edge type `e1`, by referencing `e1._src` as the starting vertex id (which, of course, this `1`) and `e1._dst` as the ending vertex id.
+This statement returns all the neighborhoods of `1` over edge type `e1`, by referencing `e1._src` as the starting vertex id (which, of course, is `1`) and `e1._dst` as the ending vertex id.
```
Comment (jude-zhu) [inline docs/manual-doc/language-structure/property-reference.md]: neighborhoods => neighbors

File: docs/manual-doc/statement-syntax/data-manipulation-statements/insert-edge-syntax.md [hunk 15]
```diff
@@ -10,7 +10,7 @@ INSERT EDGE edge_name ( <prop_name_list> ) {VALUES | VALUE}
   [ <prop_value> [, <prop_value> ] ...]
 ```
 
-INSERT EDGE statements inserts an (directed) edge from a starting vertex (given by src_vid) to an ending vertex (given by dst_vid).
+`INSERT EDGE` statements inserts an (directed) edge from a starting vertex (given by src_vid) to an ending vertex (given by dst_vid).
```
Comment (jude-zhu) [inline docs/manual-doc/statement-syntax/data-manipulation-statements/insert-edge-syntax.md]: statements => statement
an (directed) edge => a (directed) edge

File: docs/manual-doc/statement-syntax/data-manipulation-statements/pipe-syntax.md [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+
+### Examples
+
+```
+GO FROM 201 OVER edge_serve | GO FROM $-.id OVER edge_fans | GO FROM $-.id ...
+
+GO FROM 100 OVER like YIELD like._dst AS dstid, $$.player.name AS Name  | GO FROM $-.dstid OVER like YIELD like._dst, like.likeness, $-.Name
+```
+
+The dest (vertex) `id` will be given as default value if no `YIELD` is used.
```
Comment (jude-zhu) [inline docs/manual-doc/statement-syntax/data-manipulation-statements/pipe-syntax.md]: as default value => as the default value

File: docs/manual-doc/statement-syntax/data-manipulation-statements/pipe-syntax.md [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+GO FROM 201 OVER edge_serve | GO FROM $-.id OVER edge_fans | GO FROM $-.id ...
+
+GO FROM 100 OVER like YIELD like._dst AS dstid, $$.player.name AS Name  | GO FROM $-.dstid OVER like YIELD like._dst, like.likeness, $-.Name
+```
+
+The dest (vertex) `id` will be given as default value if no `YIELD` is used. 
+
+But if `YIELD` is declared explicitly, (the default value) `id` will not be given.
+
+The alias name mentioned right after placeholder `$-.` must be either exactly `id` or already defined in the previews `YIELD` statement (e.g., `distid` or `Name` as show in the above example).
```
Comment (jude-zhu) [inline docs/manual-doc/statement-syntax/data-manipulation-statements/pipe-syntax.md]: as show => as shown

File: docs/manual-doc/statement-syntax/data-manipulation-statements/yield-syntax.md [hunk 18]
```diff
@@ -24,7 +24,7 @@ nebula> YIELD "Hel" + "\tlo" AS HELLO_1, ", World!" as WORLD_2
 ----------------------
 ```
 
-* But commonly, YIELD statement is more often used to return results generated by `GO`statement. (see `GO` statement for more details).
+* But commonly, YIELD statement is used more often to return results generated by `GO`statement. (see `GO` statement for more details).
```
Comment (jude-zhu) [inline docs/manual-doc/statement-syntax/data-manipulation-statements/yield-syntax.md]: there should be a space between ''Go" and "statement".

**Comment by jude-zhu** [comment]
LGTM

**Comment by whitewum** [comment]
rebase again and again, resolve confilcts.

**Pull request merged**